### ISSUE TITLE
[Development] BDD: Add "bddQualified" To Back-End 526 Service

### ISSUE THREAD

**Issue by luke-gcio** [department-of-veterans-affairs/va.gov-team#10031]
## Issue Description

**TLDR: Add BDD qualified field to EVSS submission.  If separation date is 90-180 day from today's date true, else false.**

Decisions reached:
- Release from Active Duty (RAD) is calculated by user submission not EMIS data
- If form is submitted >180 days prior to RAD date: BE will return failure because FE should gate this. It's better for the user to simply wait to submit.
- If form is submitted < 90 days prior to RAD date: bddQualified = false.  Form is still submitted EVSS will accept the form and per Barry the form will be processed as "Quick Start," a version of 526.
- If form is submitted 180-90 days prior to RAD date: bddQualified = true.

---

Here are some notes from Michael H. We are going to be doing the calculation as to whether a service member's claim is BDD.  ~The last thing we need answered is this exam date issue which we hope will not be part of this.~

In terms of implementation, I think it makes sense that the BE pulls the submission date and the RAD (separation) date and does the calculation itself.  This was we are not pushing more business logic into the front end. 

Latest XSD attached, although I know we can pull it from the endpoint.

[526 XSD](https://app.zenhub.com/files/133843125/8ec3aadd-f908-4cca-8bbe-16ee0e206052/download)

> Weekend Update: Yesterday I deployed a new version of form526 in INT2. The XSD is attached. The JSON schema has a new element called “bddQualified”. VA.gov must set “bddQualified”=true if the service member will be release in 90 to 180 days and the SM is “available to report for examination for 45 days following submission of a disability claim.” Lighthouse must set this field to true if “BDD Program Claim” is selected on the paper 526 form.
> 
> For backwards compatibility, 526 claims will succeed if bddQualified is missing in the JSON request. However, this will cause valid BDD claims to be treated as BDD-excluded which will set the EP code to 110, 010, or 020 instead of 336.
> 
> When EVSS goes live with this change, service members with release date 1 to 89 days will be able to submit 526 requests. These should be submitted with “bddQualified”=false. The API only needs the bddQualified field when the release date is 90 to 180 days. However, this field should be sent for all requests to avoid confusion. For example, all Veteran claim submissions should have “bddQualified”=false.
> 
> Since VA.gov already allows service members with 90-180 RAD to submit 526 requests, the bddQualified field must be added to JSON requests as soon as possible after EVSS goes live with this change. We’ll be able to test BDD in PINT2 later today.

## Tasks:
- [x] Add BDD qualified field to EVSS submission.  If separation date is 90-180 day from today's date true, else false.

## How to configure this issue
- [x] **Attached to a Milestone** (when will this be completed?)
- [x] **Attached to an Epic** (what body of work is this a part of?)
- [x] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [x] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [x] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by SarahJaine** [comment]
Working on in https://github.com/department-of-veterans-affairs/vets-api/compare/sj-add-bddQualified

**Comment by annaswims** [comment]
I think we may need to have another chat about RAD date.  Right now, we're using the User-supplied RAD date for determining if the application is BDD-qualified, but otherwise using the eMIS service history to determine RAD date.  I'm thinking we want to use the same RAD date for both, but this will be a change.  Is that ok?  We will be able to start with the eMIS data in the prefilled form. 

The application expires either:
* one year and a day after the RAD date ( if rad_date > application_create_date)
* one year after the application was created (itf.creation_date > application_create_date)
* or one year from the intent to file

**Comment by annaswims** [comment]
starting to work on rad date changes

**Comment by annaswims** [comment]
Even less clarity now on what rad date should be.  Only used in ApplicationExpirationDate
#12690
https://dsva.slack.com/archives/C5AGLBNRK/p1600465023011100

**Comment by annaswims** [comment]
Closing this issue https://github.com/department-of-veterans-affairs/va.gov-team/issues/12690 more thoroughly describes the new issue.
